app/interactions/users/authenticate_user.rb

```diff
@@ -5,11 +5,6 @@ class AuthenticateUser < ActiveInteraction::Base
   string :token
 
   def execute
-    user = User.find_or_initialize_by(email: email)
-    user.first_name = first_name
-    user.last_name = last_name
-    user.token = token
-    user.save
-    user
+    User.find_by(email: email) || User.create(inputs)
```

Comment:
Great. I don't know why, but I didn't got any notification of this code that you linked, but anyway :+1: